rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1611,25 +1634,45 @@ public String getLockWhitelistAddress(int index) {
      * in the whitelist, -2 if address is invalid
      * LOCK_WHITELIST_GENERIC_ERROR_CODE otherwise.
      */
-    public Integer addLockWhitelistAddress(Transaction tx, String addressBase58, BigInteger maxTransferValue) {
+    public Integer addOneOffLockWhitelistAddress(Transaction tx, String addressBase58, BigInteger maxTransferValue) {
+        try {
+            Address address = Address.fromBase58(btcContext.getParams(), addressBase58);
+            Coin maxTransferValueCoin = Coin.valueOf(maxTransferValue.longValueExact());
+            return this.addLockWhitelistAddress(tx, new OneOffWhiteListEntry(address, maxTransferValueCoin));
+        } catch (AddressFormatException e) {
+            logger.warn(INVALID_ADDRESS_FORMAT_MESSAGE, e);
+            return -2;
+        }
+    }
+
+    public Integer addUnlimitedLockWhitelistAddress(Transaction tx, String addressBase58) {
+        try {
+            Address address = Address.fromBase58(btcContext.getParams(), addressBase58);
+            return this.addLockWhitelistAddress(tx, new UnlimitedWhiteListEntry(address));
+        } catch (AddressFormatException e) {
+            logger.warn(INVALID_ADDRESS_FORMAT_MESSAGE, e);
+            return -2;
+        }
+    }
+
+    private Integer addLockWhitelistAddress(Transaction tx, LockWhitelistEntry entry) {
         if (!isLockWhitelistChangeAuthorized(tx)) {
             return LOCK_WHITELIST_GENERIC_ERROR_CODE;
         }
 
         LockWhitelist whitelist = provider.getLockWhitelist();
 
         try {
-            Address address = Address.fromBase58(btcContext.getParams(), addressBase58);
-
-            if (whitelist.isWhitelisted(address)) {
+            if (whitelist.isWhitelisted(entry.address())) {
                 return -1;
             }
-            whitelist.put(address, Coin.valueOf(maxTransferValue.longValueExact()));
+            whitelist.put(entry.address(), entry);
             return 1;
         } catch (AddressFormatException e) {
```

Comment:
can you still get a AdressFormatExcetion even tough you removed `Address address = Address.fromBase58(btcContext.getParams(), addressBase58);` from the try catch?

Reply:
no, you can't, good catch